Update installation document of GurumDDS (#3386)
ros2/ros2_documentation: source/Installation/DDS-Implementations/Working-with-GurumNetworks-GurumDDS.rst

from the :doc:`Installing ROS 2 via Debian Packages <../Ubuntu-Install-Debians>` or
from the :doc:`Building ROS 2 on Ubuntu Linux <../Alternatives/Ubuntu-Development-Setup>`.

rmw_gurumdds requires version of GurumDDS-2.8.x.

> Debian packages of GurumDDS is provided in the ROS 2 apt repositories on linux.

Reviewer: ```suggestion
Debian packages of GurumDDS are provided in the ROS 2 apt repositories on Ubuntu.
```

Author: Applied.

Building from source code is also another way to install.

First, clone rmw_gurumdds in the ROS 2 workspace source directory.

> Don't forget to checkout the branch for the distro you want to use.

Reviewer: If I may make a suggestion: if you change the default branch of https://github.com/ros2/rmw_gurumdds to `rolling`, then instead of saying this you can just make the clone command look like:

```
git clone https://github.com/ros2/rmw_gurumdds -b {DISTRO} ros2/rmw_gurumdds
```

Author: The repo admin is on vacation, so we'll change the branch name within the next week.

Reviewer: OK, sounds good.  I'll leave this PR open until you have that done.  Please just ping me again when this is complete and we can merge this in.

Reviewer 2: > OK, sounds good. I'll leave this PR open until you have that done. Please just ping me again when this is complete and we can merge this in.

We changed the branch name, please confirm.

Reviewer: Yep, looks good.  Now you can remove the line above.  Once that is done, we can merge this.

Reviewer 2: > Yep, looks good. Now you can remove the line above. Once that is done, we can merge this.

Applied.

> rmw_gurumdds repository url is as follows: https://github.com/ros2/rmw_gurumdds

Reviewer: This sentence seems redundant, as it is in the code-block directly below.

Author: Applied.

rmw_gurumdds repository url is as follows: https://github.com/ros2/rmw_gurumdds

.. code-block:: bash

   cd ros2_ws/src
   git clone https://github.com/ros2/rmw_gurumdds ros2/rmw_gurumdds

>    git clone https://github.com/ros2/rosidl_typesupport_gurumdds ros2/rosidl_typesupport_gurumdds

Reviewer 3: this is not required anymore for rolling, humble? how about foxy?

Author: yes, we've been integrating the package so don't need it anymore for rolling, humble.
foxy is also no longer required.
